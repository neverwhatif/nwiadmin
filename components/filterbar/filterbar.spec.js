import React from 'react';
import { shallow } from 'enzyme';

import FilterBar from './index';

jest.mock('nwiadmin/components/button', () => 'Button');
jest.mock('nwiadmin/components/modalform', () => 'ModalForm');
jest.mock('nwiadmin/components/form', () => ({ Dropdown: 'Dropdown', TextInput: 'TextInput' }));
jest.mock('./filterbarcta', () => 'FilterBarCta');

const mockLocation = {
    pathname: '/lorem',
    search: '',
};

const mockHistory = {
    push: () => jest.fn(),
};

describe('FilterBar', () => {
    const component = shallow(<FilterBar location={mockLocation} history={mockHistory} filters={() => <div />} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
});
