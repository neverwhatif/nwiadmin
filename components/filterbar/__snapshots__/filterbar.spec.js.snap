// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterBar
    cta={null}
    filters={[Function]}
    hasSearch={true}
    history={
      Object {
        "push": [Function],
      }
    }
    isVisible={false}
    itemName={null}
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
    setVisibility={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <aside
        className="root"
      >
        <div
          className="search"
        >
          <TextInput
            name="search"
            onChange={[Function]}
            placeholder="Search..."
            value=""
          />
        </div>
        <div
          className="cta"
        >
          <Button
            onClick={[Function]}
          >
            Filters...
          </Button>
        </div>
        <ModalForm
          isOpen={false}
          onRequestClose={[Function]}
          onSubmit={[Function]}
          submitLabel="Apply"
          title="Search & Filter "
          type="filters"
        >
          <div />
        </ModalForm>
      </aside>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search"
          >
            <TextInput
              name="search"
              onChange={[Function]}
              placeholder="Search..."
              value=""
            />
          </div>,
          <div
            className="cta"
          >
            <Button
              onClick={[Function]}
            >
              Filters...
            </Button>
          </div>,
          <ModalForm
            isOpen={false}
            onRequestClose={[Function]}
            onSubmit={[Function]}
            submitLabel="Apply"
            title="Search & Filter "
            type="filters"
          >
            <div />
          </ModalForm>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TextInput
              name="search"
              onChange={[Function]}
              placeholder="Search..."
              value=""
            />,
            "className": "search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "search",
              "onChange": [Function],
              "placeholder": "Search...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "TextInput",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              undefined,
              <Button
                onClick={[Function]}
              >
                Filters...
              </Button>,
            ],
            "className": "cta",
          },
          "ref": null,
          "rendered": Array [
            null,
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Filters...",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Filters...",
              "type": "Button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div />,
            "isOpen": false,
            "onRequestClose": [Function],
            "onSubmit": [Function],
            "submitLabel": "Apply",
            "title": "Search & Filter ",
            "type": "filters",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "ModalForm",
        },
      ],
      "type": "aside",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <aside
          className="root"
        >
          <div
            className="search"
          >
            <TextInput
              name="search"
              onChange={[Function]}
              placeholder="Search..."
              value=""
            />
          </div>
          <div
            className="cta"
          >
            <Button
              onClick={[Function]}
            >
              Filters...
            </Button>
          </div>
          <ModalForm
            isOpen={false}
            onRequestClose={[Function]}
            onSubmit={[Function]}
            submitLabel="Apply"
            title="Search & Filter "
            type="filters"
          >
            <div />
          </ModalForm>
        </aside>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="search"
            >
              <TextInput
                name="search"
                onChange={[Function]}
                placeholder="Search..."
                value=""
              />
            </div>,
            <div
              className="cta"
            >
              <Button
                onClick={[Function]}
              >
                Filters...
              </Button>
            </div>,
            <ModalForm
              isOpen={false}
              onRequestClose={[Function]}
              onSubmit={[Function]}
              submitLabel="Apply"
              title="Search & Filter "
              type="filters"
            >
              <div />
            </ModalForm>,
          ],
          "className": "root",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TextInput
                name="search"
                onChange={[Function]}
                placeholder="Search..."
                value=""
              />,
              "className": "search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "search",
                "onChange": [Function],
                "placeholder": "Search...",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "TextInput",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                undefined,
                <Button
                  onClick={[Function]}
                >
                  Filters...
                </Button>,
              ],
              "className": "cta",
            },
            "ref": null,
            "rendered": Array [
              null,
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Filters...",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Filters...",
                "type": "Button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div />,
              "isOpen": false,
              "onRequestClose": [Function],
              "onSubmit": [Function],
              "submitLabel": "Apply",
              "title": "Search & Filter ",
              "type": "filters",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "ModalForm",
          },
        ],
        "type": "aside",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
